### PR TITLE
nodenext compatibility

### PR DESCRIPTION
**Pull request by cesarvspr**
#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by cesarvspr** [comment]
help is needed to solve types issues 

```

> @fastify/helmet@10.0.2 typescript
> tsd


  types/index.test-d.ts:106:0
  ✖  106:0  Parameter type ContentSecurityPolicy is declared too wide for argument type FastifyPluginAsync<FastifyHelmetOptions, RawServerDefault, FastifyTypeProviderDefault> & ContentSecurityPolicy.                                                                                                                                   
  ✖  198:2  Argument of type typeof fastifyHelmet is not assignable to parameter of type FastifyPluginAsync<FastifyHelmetOptions, RawServerDefault, FastifyTypeProviderDefault> & { ...; }.
  Type typeof fastifyHelmet is not assignable to type FastifyPluginAsync<FastifyHelmetOptions, RawServerDefault, FastifyTypeProviderDefault>.  

  2 errors


```

File: types/index.d.ts [hunk 1]
```diff
@@ -14,20 +14,27 @@ declare module 'fastify' {
...
   }
 
-  export interface RouteOptions extends FastifyHelmetRouteOptions {}
+  export interface RouteOptions extends fastifyHelmet.FastifyHelmetRouteOptions {}
 }
 
-export interface FastifyHelmetRouteOptions {
-  helmet?: Omit<FastifyHelmetOptions, 'global'> | false;
-}
+type FastifyHelmetPlugin = FastifyPluginCallback<fastifyHelmet.FastifyHelmetOptions>;
```
Comment (Uzlopak) [inline types/index.d.ts:20]: I think this is wrong. It is not a Callback but Async

```suggestion
type FastifyHelmetPlugin = FastifyPluginAsync<fastifyHelmet.FastifyHelmetOptions>;
```

Comment (cesarvspr) [reply types/index.d.ts]: tks for your suggestion, I think you are right

can you please help with the types issue? I already tried but could not success 

```
  types/index.test-d.ts:103:0
  ✖  103:0  Parameter type ContentSecurityPolicy is declared too wide for argument type FastifyPluginAsync<FastifyHelmetOptions, RawServerDefault, FastifyTypeProviderDefault> & ContentSecurityPolicy.  
  ✖  191:0  Parameter type FastifyPluginAsync<FastifyHelmetOptions, RawServerDefault, FastifyTypeProviderDefault> & { ...; } is declared too wide for argument type typeof fastifyHelmet.    
```

**Comment by Uzlopak** [comment]
I patched it.

**Review by climba03003 — APPROVED** [review]

**Pull request merged**